src/main/java/io/vertx/grpc/BlockingServerInterceptor.java

```diff
@@ -0,0 +1,99 @@
+package io.vertx.grpc;
+
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.vertx.core.Vertx;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Wraps the ServerInterceptor and executes it on worker thread.
+ * All incoming events will be deferred until the interception is completed.
+ *
+ * @author <a href="mailto:pkopachevskiy@corp.finam.ru">Pavel Kopachevskiy</a>
+ * @author <a href="mailto:ruslan.sennov@gmail.com">Ruslan Sennov</a>
+ */
+public class BlockingServerInterceptor implements ServerInterceptor {
+
+  public static ServerInterceptor wrap(Vertx vertx, ServerInterceptor interceptor) {
+    return new BlockingServerInterceptor(vertx, interceptor);
+  }
+
+  private final Vertx vertx;
+  private final ServerInterceptor interceptor;
+
+  private BlockingServerInterceptor(Vertx vertx, ServerInterceptor interceptor) {
+    this.vertx = vertx;
+    this.interceptor = interceptor;
+  }
+
+  @Override
+  public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> serverCall, Metadata metadata,
+                                                               ServerCallHandler<ReqT, RespT> serverCallHandler) {
+    AsyncListener<ReqT> asyncListener = new AsyncListener<>();
+    vertx.<ServerCall.Listener<ReqT>>executeBlocking(blockingCode ->
```

Comment:
does order matter? shouldn't this be unordered?

Reply:
thanks, done